### ISSUE TITLE
problem on electra's pretraining method

### ISSUE THREAD

**Issue by real-brilliant** [google-research/electra#49]
electra contains a generator and a discriminator, where generator replaces [mask] tokens with plausible alternatives ones and discriminator will figures them out.

As said in paper, 
1.  >Typically k = [0.15n], i.e., 15% of the tokens are masked out
2. > if the generator happens to generate the correct token, that token is considered “real” instead of “fake

thus only `(1 - generator_inference_acc) * 0.15` tokens is fake, and leads to a extremely unbalanced  binary classfication task for discriminator, but it does work, why?

**Comment by clarkkev** [comment]
Good question! Yes, the discriminator task is quite imbalanced, but neural nets are pretty good at fitting highly imbalanced data. One way of thinking about why the model still works well is by comparing it with BERT. BERT learns to score the correct token highly for 15% of positions. ELECTRA learns to score the correct token highly for ~94% of positions and give the other ~6% of tokens low scores. So I think ELECTRA is still receiving plenty of training signal despite the imbalanced data. However, I think it probably would help performance a bit to increase the mask percentage to even-out the class imbalance; we actually did this for the ELECTRA-Large model, using a higher mask percent of 25%.
